frontend/src/scenes/experiments/experimentLogic.tsx

```diff
@@ -923,6 +936,76 @@ export const experimentLogic = kea<experimentLogicType>([
             actions.closeSecondarySharedMetricModal()
             actions.loadExperiment()
         },
+        createExperimentDashboard: async () => {
+            actions.setIsCreatingExperimentDashboard(true)
+            try {
+                // 1. Create the dashboard
+                // 2. Create secondary metric insights in reverse order
+                // 3. Create primary metric insights in reverse order
+
+                const experimentUrl =
+                    window.location.origin + addProjectIdIfMissing(urls.experiment(values.experimentId))
+                const dashboard: DashboardType = await api.create(
+                    `api/environments/${teamLogic.values.currentTeamId}/dashboards/`,
+                    {
+                        name: 'Experiment: ' + values.experiment.name,
+                        description: `Dashboard for [${experimentUrl}](${experimentUrl})`,
+                    } as Partial<DashboardType>
+                )
+
+                // Reverse order because adding an insight to the dashboard
+                // places it at the beginning of the list
+                for (const type of ['secondary', 'primary']) {
+                    const singleMetrics =
+                        type === 'secondary' ? values.experiment.metrics_secondary : values.experiment.metrics
+                    const sharedMetrics = values.experiment?.saved_metrics.filter(
+                        (sharedMetric) => sharedMetric.metadata.type === type
+                    )
+                    const metrics = [
+                        ...singleMetrics,
+                        ...sharedMetrics.map((m) => ({ name: m.name, ...m.query })),
+                    ].reverse()
+                    if (metrics.length === 0) {
+                        return
+                    }
```

Comment:
> @andehen Hm, it seems like the code didn't execute fully. Did it reproduce multiple times? Can you look at your network requests and see what's happening there?

Here is the bug. It only works if you have both a primary _and_ a secondary metric defined here due to the early return

Reply:
Ah, good catch! That should be a continue but we can remove it entirely 612942530d